crates/core/src/network/networking.rs

```diff
@@ -105,7 +105,9 @@ impl NetworkSender for Networking {
                         to,
                         e,
                         attempts + 1,
-                        DEFAULT_MAX_CONNECT_ATTEMPTS
+                        self.retry_policy
+                            .max_attempts()
+                            .unwrap_or(NonZeroUsize::MAX), // always Some, unreachable by RetryPolicy::None
```

Comment:
What does the comment refer to? That `RetryPolicy::max_attempts()` is always `Some`? I think one can configure a fixed delay retry policy with no maximum number of attempts.

Reply:
If we use a `RetryPolicy::None`, this code is unreachable. The fact that we're in this branch means that we have a policy with _some_ retry attempts.

Reply:
I think you can have something like this: `RetryPolicy::fixed_delay(Duration::from_millis(10), None).max_attempts() == None`. So not `RetryPolicy::None` policies with infinite retries. The code is correct. Just the comment is a bit confusing.

Reply:
Gotcha! Updated :-)